src/python/gem5/prebuilt/viper/board.py

```diff
@@ -0,0 +1,191 @@
+        # Enables AVX, OSXSAVE, XSAVE, POPCNT, SSE4.2, SSE4.1, CMPXCHG16B,
+        # and FMA.
+        avx_cpu_features = [0x00020F51, 0x00000805, 0xEFDBFBFF, 0x1C803209]
+        for core_wrapper in self.get_processor().get_cores():
+            if not core_wrapper.is_kvm_core():
+                warn("AVX is only supported with KVM cores")
+            core = core_wrapper.get_simobject()
+            for isa in core.isa:
+                isa.vendor_string = "AuthenticAMD"
+                isa.ExtendedState = avx_extended_state
+                isa.FamilyModelStepping = avx_cpu_features
+
+        # The System() object in gem5 has a memories parameter which defaults
+        # to Self.all. This will collect *all* AbstractMemories and connect to
+        # the CPU side. To avoid this we manually assign the memories param to
+        # the CPU side memories. We need the MemInterface which is called dram
+        # in the MemCtrl class even though it might not be modelling dram.
+        self.memories = self.memory.get_mem_interfaces()
+
+    def get_pci_host(self):
+        return self.pc.pci_host
+
+    @overrides(KernelDiskWorkload)
+    def get_disk_device(self):
+        return "/dev/sda"
+
+    @overrides(KernelDiskWorkload)
+    def get_default_kernel_args(self) -> List[str]:
+        # The regular parameters used with gem5 plus (1) fbdev_emulation=0
+        # to disable having to implement this functionality, (2) blacklist
+        # amdgpu because we need to copy the VBIOS into memory first, and (3)
+        # blacklist psmouse as amdgpu driver adds new mouse commands which
+        # gem5 does not implement and they do not seem to be documented.
+        return [
+            "earlyprintk=ttyS0",
+            "console=ttyS0",
+            "lpj=7999923",
+            "root={root_value}",
+            "drm_kms_helper.fbdev_emulation=0",
+            "modprobe.blacklist=amdgpu",
+            "modprobe.blacklist=psmouse",
+        ]
+
+    @overrides(KernelDiskWorkload)
+    def _set_readfile_contents(self, readfile_contents):
+        """In the case of a GPU workload, we need to load the GPU driver first.
+        This method will prepend the driver load command to the
+        readfile_contents. Note that this assumes that all of the gpus are the
+        same, if there's more that one GPU.
+        """
+
+        if self.get_devices() is None:
+            warn("No GPU devices. Not loading GPU driver.")
+            return super()._set_readfile_contents(readfile_contents)
+
+        driver_load_command = self.get_devices()[0].get_driver_command()
+        contents = driver_load_command + "\n" + readfile_contents
+
+        return super()._set_readfile_contents(contents)
+
+    # Replicate the capability of the old GPUFS config, which embed a binary
+    # application or script into a bash script setting up the environment and
+    # loading the GPU driver.
+    def make_gpu_app(self, gpu: BaseViperGPU, app: str, debug: bool = False):
```

Comment:
Here, can you override `set_binary_to_run`? Also, `self` should have a reference to the `gpu`, right? Maybe use `gpu[0]` and warn or document that GPUs must be the same GPU architecture.

Reply:
I am not sure how to do that since this function takes `str` not `BinaryResource`.  As far as I can tell only your team can create resources, right?  The GPU resources currently available are about 3 years old.  They don't know what an MI300X is and in GPU development time they may as well be a century old.  So I don't have a way to test this either